### PR TITLE
Extend secrets management and reorganize `continuous-delivery` folders 

### PR DESCRIPTION
**Pull request by EliiseS**
# Pull Request Template

## What are you trying to address

Extending the information available on managing secret with gitops.

## Description of new changes

- Reorganize folder structure for `/continuous-delivery` to group together topics by discipline, as discussed with @TessFerrandez 
- Adding `Secret Management with GitOps`  a heading on External Secret Provider and additional minor information to the other headings. 

Co-author: @buzzfrog 

## For all pull requests

- [ ] Link to the issue you are solving (so it gets closed)
- [ ] Label the pull request with the appropriate area(s)
- [ ] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [ ] Assign the project - Engineering Playbook Backlog
- [ ] Assign the pull request to yourself

### CODE REVIEW

**Comment by EliiseS** [comment]
Can someone help me understand what this pipeline error is about? I can't tell if I caused it:

image

**Comment by shiranr** [comment]
Hi @EliiseS,
The error is in the upper log.
I think the issue is the links which point to the documents you moved.
Screenshot 2023-01-18 at 14 31 27
LMK if you need more assistance.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
+# Secret Management with GitOps
+
+GitOps primarily have git repositories in the center considered a source of truth for managing both infrastructure and application. This Infrastructure and application will require secured access to other resources of the system through secrets. Committing clear text secrets into git repositories is unacceptable even if the repositories are private to your team and organization. Teams need a secure way to handle secrets when using GitOps.
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:3]: A few suggestions to consider:
* GitOps primarily --> GitOps <something> (projects | solutions | etc?) primarily... (also, is the "primarily" qualifier needed?)
* "This Infrastructure" --> lower-case infrastructure
* "clear text" --> "clear-text". 
* Consider expanding briefly on why committing secrets into repositories is a "bad thing™️"

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+There are many ways to manage secrets with GitOps and at high level can be categorized into:
+
+1. Encrypted Secrets in Git Repositories
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:7]: Inconsistent capitalization between 1 & 2.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+1. Reference to secrets stored in the external key vault
+
+## 1. Encrypted Secrets in Git Repositories
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:10]: Capitalization is also inconsistent here between 1 & 2. Is it consistent in this particular playbook to use numbered indexes for H2's? Fine if it is normally done - it is just unusual coming from a different playbook, so figured it was worth a mention.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+In this approach, secrets are manually encrypted by developers using a public key and the custom Kubernetes controller running in the target cluster can only decrypt the key. Some popular tools for his approach are [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets), [Mozilla SOPS](https://github.com/mozilla/sops)
+
+- Secret changes are managed by making changes within the GitOps repository which provides great traceability of the changes made
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:14]: It may be useful to include a sentence leading into these bullets to provide context, such as "Benefits of using this approach include:"

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+- All secrets can be rotated by making changes in GitOps, without accessing the cluster
+
+### [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets)
+
+Sealed secrets use asymmetric encryption to encrypt secrets. Kubernetes controller generates key pair(private-public) and stores the private key in etcd as Kubernetes secret. Developers use [Kubeseal CLI](https://github.com/bitnami-labs/sealed-secrets#public-key--certificate) to seal secrets before committing to the git repo.
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:19]: Consider a space between "pair" and the parenthesis. Would it read better as "A Kubernetes controller generates a key-pair (private-public) and stores the private key in the cluster's etcd database as a Kubernetes secret."

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+- [Support automatic key rotation](https://github.com/bitnami-labs/sealed-secrets#sealing-key-renewal) for the private key and can be used to enforce re-encryption of secrets
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:21]: Similar to above, a lead-in to provide context for the following list may be helpful.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+  - Due to [automatic renewal of the sealing key](https://github.com/bitnami-labs/sealed-secrets#sealing-key-renewal), the key needs to be prefetched from the cluster or cluster set up to store the sealing key on renewal in a secondary location
+- Multi-tenancy support at the namespace level can be enforced by the controller
+- When sealing secrets developers needs a connection to the cluster control plane to fetch the public key or the public key has to be explicitly shared with the developer
+- Potential risk of secret checked into a git repo
+- If the private key in the cluster is lost for some reason all secrets need to be re-encrypted followed by a new key-pair generation
+- Does not scale with Multi-cluster here every cluster will require a controller having its own key pair
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:27]: Does "Multi" need to be capitalized?
Should there be a hyphen or a period between "cluster" and "here"?

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+
+### [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets)
+
+Sealed secrets use asymmetric encryption to encrypt secrets. Kubernetes controller generates key pair(private-public) and stores the private key in etcd as Kubernetes secret. Developers use [Kubeseal CLI](https://github.com/bitnami-labs/sealed-secrets#public-key--certificate) to seal secrets before committing to the git repo.
+
+- [Support automatic key rotation](https://github.com/bitnami-labs/sealed-secrets#sealing-key-renewal) for the private key and can be used to enforce re-encryption of secrets
+  - Due to [automatic renewal of the sealing key](https://github.com/bitnami-labs/sealed-secrets#sealing-key-renewal), the key needs to be prefetched from the cluster or cluster set up to store the sealing key on renewal in a secondary location
+- Multi-tenancy support at the namespace level can be enforced by the controller
+- When sealing secrets developers needs a connection to the cluster control plane to fetch the public key or the public key has to be explicitly shared with the developer
+- Potential risk of secret checked into a git repo
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:25]: Should this be expended on a little bit?

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+- If the private key in the cluster is lost for some reason all secrets need to be re-encrypted followed by a new key-pair generation
+- Does not scale with Multi-cluster here every cluster will require a controller having its own key pair
+- Secrets are stored encrypted in the gitops repository, if the private encryption key is leaked, all secrets can be decrypted
+
+### [Mozilla SOPS](https://github.com/mozilla/sops)
+
+SOPS: Secrets OPerationS is an encryption tool that supports YAML, JSON, ENV, INI, and BINARY formats and encrypts with AWS KMS, GCP KMS, Azure Key Vault, age, and PGP and is not just limited to Kubernetes. It supports integration with some common key management systems including Azure Key Vault. Where the key management system is used to store the encryption key for encrypting secrets and not the actual secrets.
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:32]: Is the clause "Where the key management system..." part of the preceding sentence? (Consider removing the period after Key Vault?)

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+
+- [Flux](https://toolkit.fluxcd.io/guides/mozilla-sops/#azure) has better support for SOPS with cluster-side decryption.
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:34]: Similar to above, a lead-in to provide context for the following list may be helpful.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+- An added layer of security as the private key used for decryption is protected in an external key vault.
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:35]: Missing verb compared to the other items in the list. Perhaps "Providers an added layer..."?

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+- Needs plugin to work with Helm([Helm Secrets](https://github.com/jkroepke/helm-secrets)) and Kustomization ([KSOPS](https://github.com/viaduct-ai/kustomize-sops))([kustomize-sopssecretgenerator](https://github.com/goabout/kustomize-sopssecretgenerator))
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:36]: Space after Helm?

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+- Does not scale with larger teams as each developer has to encrypt the secrets
+- The public key is sufficient for creating brand new files. The secret key is required for decrypting and editing existing files because SOPS computes a MAC on all values.  When using the public key solely to add or remove a field, the whole file should be deleted and recreated.
+
+## 2. Reference to secrets stored in an external key vault (recommended)
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:40]: Consider adding a sentence that also indicates this is the recommended solution following the bulleted list that introduces these above (will repeat this comment up there.)

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 14]
```diff
@@ -0,0 +1,177 @@
+# Secret Management with GitOps
+
+GitOps primarily have git repositories in the center considered a source of truth for managing both infrastructure and application. This Infrastructure and application will require secured access to other resources of the system through secrets. Committing clear text secrets into git repositories is unacceptable even if the repositories are private to your team and organization. Teams need a secure way to handle secrets when using GitOps.
+
+There are many ways to manage secrets with GitOps and at high level can be categorized into:
+
+1. Encrypted Secrets in Git Repositories
+1. Reference to secrets stored in the external key vault
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:8]: Consider adding a sentence following this list such as "Referencing secrets in an external key vault is the recommended approach", or otherwise including that recommendation up here.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+## Important Links
+
+- [Sealed Secrets with Flux v2](https://toolkit.fluxcd.io/guides/sealed-secrets/)
+- [Mozilla SOPS with Flux v2](https://toolkit.fluxcd.io/guides/mozilla-sops/)
+- [Secret Management with Argo CD](https://argo-cd.readthedocs.io/en/stable/operator-manual/secret-management/)
+- [Secret management Workflow](https://www.youtube.com/watch?v=-k6HEXaE75k)
+
+## Appendix
+
+### Secrets Rotation on Environment Variables and Mounted Secrets
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:117]: Consider this as a standalone article? Maybe either as part of this PR or as a subsequent effort post-merge of this PR?

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+
+## 2. Reference to secrets stored in an external key vault (recommended)
+
+This approach relies on a key management system like [Azure Key Vault](https://learn.microsoft.com/en-us/azure/key-vault/general/overview) to hold the secrets and the git manifest in the repositories has reference to the key vault secrets. Developers do not perform any cryptographic operations with files in repositories. Kubernetes operators running in the target cluster are responsible for pulling the secrets from the key vault and making them available either as Kubernetes secrets or secrets volume mounted to the pod.
+
+For secret rotation ideas, see [Secrets Rotation on Environment Variables and Mounted Secrets](#secrets-rotation-on-environment-variables-and-mounted-secrets)
+
+For how to authenticate private container registries with a Service principal see: [Authenticated Private Container Registry](#authenticated-private-container-registry)
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:46]: lower caps for "Service principal" --> "service principal". Might be better to call it an "Azure Active Directory (Azure AD) service principal".

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+
+All the below options provide the following:
+
+- Secrets are not stored in the repository
+- Supports Prometheus metrics for observability
+- Supports sync with Kubernetes Secrets
+- Supports Linux and Windows containers
+- Provides enterprise-grade external secret management
+- Easily scalable with multi-cluster and larger teams
+- Both solutions support either Service Principal or Managed Identity for [authentication with the Key Vault](https://learn.microsoft.com/en-us/azure/key-vault/general/authentication).
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:56]: Lower-case for "service principal" and "managed identity". It might be useful to include links to MI's and SP's when introducing the concepts. https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+
+### [Azure Key Vault Provider for Secrets Store CSI Driver](https://github.com/Azure/secrets-store-csi-driver-provider-azure)
+
+Azure Key Vault Provider (AKVP) for Secrets Store CSI Driver is an azure vault provider for [Kubernetes secret store CSI Driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver) allows you to get secret content stored in Azure Key vault instance and use the Secrets Store CSI driver interface to mount them into Kubernetes pods. Mounts secrets/keys/certs to pod using a CSI Inline volume.
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:60]: Sentence is awkward. Maybe "...that allows you to..."? Not sure...

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 19]
```diff
@@ -0,0 +1,177 @@
...
+
+Azure Key Vault Provider for Secrets Store CSI Driver [install guide](https://secrets-store-csi-driver.sigs.k8s.io/getting-started/installation.html).
+
+CSI driver will need access to Azure Key vault either through  or managed identity(recommended). To make this access secure you can leverage [Azure Workload Identity](https://github.com/Azure/azure-workload-identity)(recommended) or [AAD Pod Identify](https://github.com/Azure/aad-pod-identity). Please note AAD pod identity will soon be replaced by Azure WorkLoad Identity.
```
Comment (John-Garland) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:64]: Missing word(s) - "either through <space> or managed identity"
Azure Workload Identity --> Azure AD Workload Identity (and in later one, the L in WorkLoad should be lower case)

**Review by John-Garland — CHANGES_REQUESTED** [review]
Was able to review until start of Key Vault section. Will share link with others and also try to review remaining sections a bit later, but sharing content for now.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 1 above)
Comment (TessFerrandez) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:3]:
```suggestion
GitOps primarily have git repositories in the center considered a source of truth for managing both infrastructure and application. This infrastructure and application will require secured access to other resources of the system through secrets. Committing clear-text secrets into git repositories is unacceptable even if the repositories are private to your team and organization. Teams need a secure way to handle secrets when using GitOps.
```

Comment (TessFerrandez) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:3]: @John-Garland a tip when you review... if you use suggestions like above, the author can click on commit suggestion to easily incorporate your comments... (it's the (+-) button when you add a comment)

Comment (EliiseS) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md]: Should I clarify why committing clear texts secrets is a bad practice or commiting secrets in general?

Comment (EliiseS) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md]: I chose to believe you meant the former and tried my best in providing an example. Let me know if you'd word it differently 😄

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 8 above)
Comment (EliiseS) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:25]: Elaborated on in the above bulleted pointed section

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 13 above)
Comment (EliiseS) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:40]: You'd like to add that only at the top, right? This is just a duplicate comment?

Comment (John-Garland) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:40]: Yes.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 18 above)
Comment (EliiseS) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:60]: I see your point, reworded

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 19 above)
Comment (EliiseS) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:64]: Tsk, tsk. You're losing your edge. You missed the misspelled `identify-> identity`

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 15 above)
Comment (EliiseS) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:124]: @TessFerrandez What would you suggest?

**Review by shiranr — APPROVED** [review]
Very great content. LGTM.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 3 above)
Comment (TessFerrandez) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:12]: I would recommend not numbering them in this case... numbering suggests that there is an order to things, and in this case it is just two different categories without an internal order, so we should probably remove the numbering both for the titles and the list

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 20]
```diff
@@ -0,0 +1,189 @@
+# Secret Management with GitOps
+
+GitOps projects have git repositories in the center that are considered a source of truth for managing both infrastructure and application. This infrastructure and application will require secured access to other resources of the system through secrets. Committing clear-text secrets into git repositories is unacceptable even if the repositories are private to your team and organization. This is unacceptable for a multitude of reasons, but an example can be because secrets are then stored in clear-text on every developers computer and if a developers computer is compromised, so are all the secrets. Teams need a secure way to handle secrets when using GitOps.
```
Comment (TessFerrandez) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:3]:
```suggestion
GitOps projects have git repositories in the center that are considered a source of truth for managing both infrastructure and application. This infrastructure and application will require secured access to other resources of the system through secrets. Committing clear-text secrets into git repositories is unacceptable even if the repositories are private to your team and organization.  Teams need a secure way to handle secrets when using GitOps.
```

Comment (TessFerrandez) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:3]: I think we can skip the last sentence, to make it crisper, it should be obvious to everyone why we should not store secrets in repos.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 21]
```diff
@@ -0,0 +1,189 @@
...
+
+There are many ways to manage secrets with GitOps and at high level can be categorized into:
+
+1. Encrypted secrets in git repositories
+1. Reference to secrets stored in the external key vault
+
+> **TLDR**: Referencing secrets in an external key vault is the recommended approach, because of, but not limited to, storing encrypted secrets in repo can result in secrets being exposed if the private key is leaked, easier to orchestrate secret rotation and not as scalable to with multiple clusters or teams.
```
Comment (TessFerrandez) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:10]:
```suggestion
> **TLDR**: Referencing secrets in an external key vault is the recommended approach. If the private key is leaked, storing it in key vault makes it easy to orchestrate secret rotation.
```

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 22]
```diff
@@ -0,0 +1,189 @@
...
+
+## 1. Encrypted Secrets in Git Repositories
+
+In this approach, secrets are manually encrypted by developers using a public key and the custom Kubernetes controller running in the target cluster can only decrypt the key. Some popular tools for his approach are [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets), [Mozilla SOPS](https://github.com/mozilla/sops)
```
Comment (TessFerrandez) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:14]:
```suggestion
In this approach, Developers manually encrypt secrets using a public key, and the key can only be decrypted by the custom Kubernetes controller running in the target cluster. Some popular tools for his approach are [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets), [Mozilla SOPS](https://github.com/mozilla/sops)
```

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 23]
```diff
@@ -0,0 +1,189 @@
...
+
+All the secret encryption tools share the following:
+
+- Secret changes are managed by making changes within the GitOps repository which provides great traceability of the changes made
```
Comment (TessFerrandez) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:18]:
```suggestion
- Secret changes are managed by making changes within the GitOps repository which provides great traceability
```

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 24]
```diff
@@ -0,0 +1,189 @@
...
+- All secrets can be rotated by making changes in GitOps, without accessing the cluster
+- Secrets are stored encrypted in the gitops repository, if the private encryption key is leaked, all secrets can be decrypted
```
Comment (TessFerrandez) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:20]: This sounds scary

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 25]
```diff
@@ -0,0 +1,189 @@
...
+- Supports fully disconnected gitops scenarios
```
Comment (TessFerrandez) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:21]:
```suggestion
- They support fully disconnected gitops scenarios
```

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 15 above)
Comment (TessFerrandez) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:129]: I agree wit that, this page is getting a bit long so it would be nice to break this out to a separate page

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md [hunk 26]
```diff
@@ -0,0 +1,189 @@
...
+- [Sealed Secrets with Flux v2](https://toolkit.fluxcd.io/guides/sealed-secrets/)
+- [Mozilla SOPS with Flux v2](https://toolkit.fluxcd.io/guides/mozilla-sops/)
+- [Secret Management with Argo CD](https://argo-cd.readthedocs.io/en/stable/operator-manual/secret-management/)
+- [Secret management Workflow](https://www.youtube.com/watch?v=-k6HEXaE75k)
+
+## Appendix
+
+### Secrets Rotation on Environment Variables and Mounted Secrets
+
+1. Mapping Secrets via secretKeyRef with environment variables.
```
Comment (TessFerrandez) [inline docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:131]: Again, same comment about the numbering, the numbering makes it look like a step by step recipe while these are really just different options

**Review by TessFerrandez — APPROVED** [review]
Thank you much for the contribution!!!
Approving with language and restructuring suggestions

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 20 above)
Comment (EliiseS) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:3]: I am more than happy to do this, however you must know your comment directly conflicts with @John-Garland's on the same section 😄

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 24 above)
Comment (EliiseS) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:21]: This is what it means to have encrypted secrets in the repository and is one the downsides of course.

File: docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md (hunk 21 above)
Comment (EliiseS) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:11]: I'm not sure I understand the text after your suggestion. ` If the private key is leaked, storing it in key vault makes it easy to orchestrate secret rotation.` When referencing the key vault, there is no private key that needs to be managed, because that's being handled by the controller on the cluster.

Comment (TessFerrandez) [reply docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md:11]: :) I understand now, I could not understand the text in the original context, which is why I wanted to change it to something that is more understandable. 

We should change it to something akin to "Referencing secrets in an external key vault is the recommended approach. If the secret is leaked, storing it in key vault makes it easy to orchestrate secret rotation."

**Review by John-Garland — APPROVED** [review]
Some capitalization issues remaining ("service principal" is capitalized in a couple places.) Nothing worth stopping progress, though.
